### PR TITLE
Add interval calibration to the bitcoin metrics

### PR DESCRIPTION
**Pull request by IvanIvanoff**
#### Summary
Clean the rest of the calibrations
Make the calibrations more consistent in the number of data points returned

[//]: # (#### Related PRs)

[//]: # (#### Additional deploy notes)

[//]: # (#### Screenshots)

### CODE REVIEW

File: lib/sanbase_web/graphql/resolvers/clickhouse_resolver.ex [hunk 1]
```diff
@@ -93,7 +93,10 @@ defmodule SanbaseWeb.Graphql.Resolvers.ClickhouseResolver do
   end
 
   def mvrv_ratio(_root, %{slug: "bitcoin", from: from, to: to, interval: interval}, _resolution) do
-    Bitcoin.mvrv_ratio(from, to, interval)
+    with {:ok, from, to, interval} <-
+           calibrate_interval(Bitcoin, "bitcoin", from, to, interval, 86400, 50) do
```
Comment (kalkov) [inline lib/sanbase_web/graphql/resolvers/clickhouse_resolver.ex]: Can you extract to constants `86_400` & `50`?

File: lib/sanbase/clickhouse/bitcoin/bitcoin.ex [hunk 2]
```diff
@@ -158,4 +150,12 @@ defmodule Sanbase.Clickhouse.Bitcoin do
 
     {query, args}
   end
+
+  # Round the interval to whole days only
+  defp round_interval(interval) when is_binary(interval) do
+    Sanbase.DateTimeUtils.compound_duration_to_seconds(interval)
+    |> div(86_400)
```
Comment (kalkov) [inline lib/sanbase/clickhouse/bitcoin/bitcoin.ex]: This can go into constant

File: lib/sanbase_web/graphql/resolvers/etherbi_resolver.ex [hunk 3]
```diff
@@ -20,38 +28,29 @@ defmodule SanbaseWeb.Graphql.Resolvers.EtherbiResolver do
         %{slug: "bitcoin", from: from, to: to, interval: interval},
         _resolution
       ) do
-    Bitcoin.token_age_consumed(from, to, interval)
+    with {:ok, from, to, interval} <-
+           calibrate_interval(Bitcoin, "bitcoin", from, to, interval, 86400, 50) do
```
Comment (kalkov) [inline lib/sanbase_web/graphql/resolvers/etherbi_resolver.ex]: Can be taken from constants

File: lib/sanbase_web/graphql/resolvers/etherbi_resolver.ex [hunk 4]
```diff
@@ -68,26 +67,26 @@ defmodule SanbaseWeb.Graphql.Resolvers.EtherbiResolver do
...
-             Blockchain.TokenAgeConsumed,
-             contract_address,
+           calibrate_interval(
+             TokenAgeConsumed,
+             contract,
              from,
              to,
              interval,
-             60 * 60,
+             3600,
```
Comment (kalkov) [inline lib/sanbase_web/graphql/resolvers/etherbi_resolver.ex]: We have `@one_hour_seconds 3600` in clickhouse_resolver, you can extract it here as well.

File: lib/sanbase/clickhouse/bitcoin/bitcoin.ex (hunk 2 above)
Comment (IvanIvanoff) [reply lib/sanbase/clickhouse/bitcoin/bitcoin.ex]: having a constant for as single value like here will reduce the readability imo

File: lib/sanbase_web/graphql/resolvers/etherbi_resolver.ex (hunk 4 above)
Comment (IvanIvanoff) [reply lib/sanbase_web/graphql/resolvers/etherbi_resolver.ex]: I don't see much logic in using it. I prefer to use constants when they can change like the number of datapoints returned. The seconds in an hour won't change

File: lib/sanbase_web/graphql/resolvers/clickhouse_resolver.ex (hunk 1 above)
Comment (IvanIvanoff) [reply lib/sanbase_web/graphql/resolvers/clickhouse_resolver.ex]: They are not constants and can vary between queries

Comment (kalkov) [reply lib/sanbase_web/graphql/resolvers/clickhouse_resolver.ex]: It's not obvious what 86400 is. Also 50. One way to make it more explicit is by extracting into constant, the other - using named params. The first one is much simpler in the current case.

File: lib/sanbase/clickhouse/bitcoin/bitcoin.ex (hunk 2 above)
Comment (kalkov) [reply lib/sanbase/clickhouse/bitcoin/bitcoin.ex]:
```elixir
Sanbase.DateTimeUtils.compound_duration_to_seconds(interval)
    |> div(@one_day_in_seconds)
```
seems more readable to me

File: lib/sanbase_web/graphql/resolvers/clickhouse_resolver.ex (hunk 1 above)
Comment (IvanIvanoff) [reply lib/sanbase_web/graphql/resolvers/clickhouse_resolver.ex]: If you think this 86400 is so not understandable let's have a card for replacing it as we have in 12-15 places and changing just one and ignoring the others won't do any good

**Pull request merged**